R/summarystatsbayesianztest.R

```diff
@@ -0,0 +1,404 @@
+  summaryTable$addColumnInfo(name = "bf",    title = bfTitle,                       type = "number")
+  summaryTable$addColumnInfo(name = "maxBf", title = gettextf("Max %1$s", bfTitle), type = "number")
+
+  jaspResults[["summaryTable"]] <- summaryTable
+
+  data <- jaspResults[["data"]][["object"]]
+  bf10 <- .bayesianZTestsComputeBf10(
+    x   = data[["es"]],
+    se  = data[["se"]],
+    mu  = options[["priorMean"]],
+    sigma = options[["priorSd"]],
+    alternative = switch(
+      options[["hypothesis"]],
+      "notEqualToTestValue"  = "two.sided",
+      "greaterThanTestValue" = "greater",
+      "lessThanTestValue"    = "less"
+    ))
+  maxBf <- 1/exp(-(data[["es"]]/data[["se"]])^2/2)
+
+
+  bf10  <- .recodeBFtype(bf10, options[["bayesFactorType"]])
+  maxBf <- .recodeBFtype(maxBf, options[["bayesFactorType"]])
+
+  newRow <- list(
+    es    = data[["es"]],
+    se    = data[["se"]],
+    bf    = bf10,
+    maxBf = maxBf
+  )
+
+  if (options[["defaultBf"]] && options[["defaultBfType"]] != "select") {
+
+    summaryTable$addColumnInfo(name = "defBf", title = gettextf("Default %1$s", bfTitle), type = "number")
+
+    defBf <- .bayesianZTestsGetDefaultBf(data[["es"]], data[["se"]], options)
+    defBf <- .recodeBFtype(defBf, options[["bayesFactorType"]])
+
+    newRow$defBf <- defBf
+    summaryTable$addFootnote(.bayesianZTestsGetDefaultBfNote(options))
+  }
+
+
+  summaryTable$addRows(newRow)
+
+  return()
+}
+.bayesianZTestPriorPosteriorPlot <- function(jaspResults, options) {
+
+  if (!is.null(jaspResults[["priorPosteriorPlot"]]))
+    return()
+
+  priorPosteriorPlot <- createJaspPlot(title = gettext("Prior Robustness Plot"), width = 450, height = 300)
+  priorPosteriorPlot$position <- 2
+  priorPosteriorPlot$dependOn(c(.bayesianZTestsDependencies, c("plotPriorAndPosterior", "plotPriorAndPosteriorAdditionalInfo")))
+  jaspResults[["priorPosteriorPlot"]] <- priorPosteriorPlot
+
+  data      <- jaspResults[["data"]][["object"]]
+  alt       <- switch(
+    options[["hypothesis"]],
+    "notEqualToTestValue"  = "two.sided",
+    "greaterThanTestValue" = "greater",
+    "lessThanTestValue"    = "less"
+  )
+
+  BF10 <- .bayesianZTestsComputeBf10(
+    x  = data[["es"]], se = data[["se"]],
+    mu = options[["priorMean"]], sigma = options[["priorSd"]],
+    alternative = alt
+  )
+  plotData <- .bayesianZTestsComputePlot(
+    x  = data[["es"]], se = data[["se"]],
+    mu = options[["priorMean"]], sigma = options[["priorSd"]],
+    alternative = alt,
+    isLog = options[["dataType"]] == "esAndCiLog"
+  )
+
+  plot <- jaspGraphs::PlotPriorAndPosterior(
+    dfLines    = plotData[["dfLines"]],
+    dfPoints   = plotData[["dfPoints"]],
+    BF         = if(options[["plotPriorAndPosteriorAdditionalInfo"]]) {if (options[["bayesFactorType"]] == "BF01") 1/BF10 else BF10},
+    bfType     = if(options[["plotPriorAndPosteriorAdditionalInfo"]]) {if (options[["bayesFactorType"]] == "BF01") "BF01" else "BF10"},
```

Comment:
how about
```suggestion
    BF         = if(options[["plotPriorAndPosteriorAdditionalInfo"]] && options[["bayesFactorType"]] == "BF01") { 1/BF10 else BF10},
    bfType     = if(options[["plotPriorAndPosteriorAdditionalInfo"]] && options[["bayesFactorType"]] == "BF01") { "BF01" else "BF10"},
```
?

Reply:
I think that those two are actually different.

My returns null when ``options[["plotPriorAndPosteriorAdditionalInfo"]]`` is FALSE and selects one of the remaining options based on ``options[["bayesFactorType"]]``

Yours does something oddly different? (also the ``else`` within the curly brackets?)

Reply:
> My returns null when options[["plotPriorAndPosteriorAdditionalInfo"]] is FALSE

Ah you're right, I missed that.